packages/dara-components/js/common/select/select.tsx

```diff
@@ -169,7 +173,10 @@ function Select(props: SelectProps): JSX.Element {
             />
         );
     }
-    const [selectedItem, setSelectedItem] = useState(itemArray.find((item) => String(item.value) === String(value)));
+    const explicitValue = toItem(value);
+    const [selectedItem, setSelectedItem] = useState(
+        explicitValue ?? itemArray.find((item) => String(item.value) === String(value))
```

Comment:
Should this condition not be reversed? i.e. use the item from the list and fall back to the explicit one